examples/write_smart_contract.rs

```diff
@@ -0,0 +1,50 @@
+use ethers::{core::{abi::Abi,types::{Address, H256}}};
+use ethers::middleware::SignerMiddleware;
+use ethers::contract::Contract;
+use ethers::providers::{Provider, Http};
+use ethers::signers::{LocalWallet, Signer};
+use std::{convert::TryFrom, sync::Arc};
+use eyre::Result;
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    
+   // this wallet's private key
+   let wallet = "<private key>".parse::<LocalWallet>()?;
+   
+   // (ugly way to write the ABI inline, you can otherwise read it from a file)
+   let abi: Abi = serde_json::from_str(r#"<contract's abi>"#)?;
```

Comment:
this and the pk will of course fail.

if this "example" is rather a template, then this is fine.

But we should name this example accordingly and move this to a separate function so it's not run by default